src/ffi/pystate.rs

```diff
@@ -1,13 +1,17 @@
 #[cfg(not(PyPy))]
 use crate::ffi::moduleobject::PyModuleDef;
 use crate::ffi::object::PyObject;
-use std::os::raw::c_int;
-#[cfg(not(PyPy))]
-use std::os::raw::c_long;
+use std::os::raw::{c_int, c_long};
 
 pub const MAX_CO_EXTRA_USERS: c_int = 255;
 
-opaque_struct!(PyThreadState);
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct PyThreadState {
+    pub ob_base: PyObject,
+    pub interp: *mut PyInterpreterState,
+}
+
```

Comment:
Can you show me where you got this from?

I removed this exact definition in #1787, as I understood it to be incorrect. It looked to me like the corresponding struct in the C header is super complex and I didn't think it was worth trying to wrap.

https://github.com/python/cpython/blob/e41912c6348362489d2514565a70782591f23902/Include/cpython/pystate.h#L73